src/components/ArticleList.jsx

```diff
@@ -0,0 +1,40 @@
+import { useState, useEffect } from "react";
+import { getAllArticles } from "./apiCalls";
+import { ArticleCard } from "./ArticleCard";
+
+export const ArticleList = () => {
+  const [articles, setArticles] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getAllArticles()
+      .then(({ articles }) => {
+        setIsLoading(false);
+        setArticles(articles);
+      })
+      .catch((err) => {
+        setIsLoading(false);
+        setIsError(true);
+      });
+  }, []);
+
+  if (isLoading) {
+    return <p>Loading...</p>;
+  }
+
+  if (isError) {
+    return <p>Bad request</p>;
```

Comment:
Don't have to but we can now be more descriptive about errors on the front end as it will be what a user sees.